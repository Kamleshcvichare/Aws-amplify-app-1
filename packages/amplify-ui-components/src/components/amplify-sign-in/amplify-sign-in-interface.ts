export interface AmplifyForgotPasswordHintProps {
  forgotPasswordText: string;
  resetPasswordText: string;
  authState?: string;
  onAuthStateChange?: any;
}

export interface AmplifySignInFormFooterProps {
  submitButtonText: string;
  noAccountText: string;
  createAccountText: string;
  overrideStyle?: boolean;
  authState?: string;
  onAuthStateChange?: any;
}