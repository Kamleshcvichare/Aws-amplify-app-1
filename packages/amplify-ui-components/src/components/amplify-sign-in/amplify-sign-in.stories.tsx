import { h } from '../../common/jsx2dom';
import { knobs } from '../../common/testing';

export default {
  title: 'amplify-sign-in',
};

export const withOverrideStyle = () => <amplify-sign-in overrideStyle={knobs.overrideStyleKnob()}></amplify-sign-in>;
