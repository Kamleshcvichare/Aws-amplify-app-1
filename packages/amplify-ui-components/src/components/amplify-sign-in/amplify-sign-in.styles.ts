import { css } from 'emotion';

export const forgotPasswordLink = css`
  font-family: var(--font-family);
  font-weight: lighter;
  font-size: 14px;
  margin-bottom: 	2.625rem;

  a:link {
    color: var(--amazon-orange);
    text-decoration: none;
  }

  a:hover {
    color: var(--light-amazon-orange);
    text-decoration: underline;
  }
  
  a:active {
    color: var(--button-click);
    text-decoration: underline;
  }
`;

export const signInFormFooter = css`
  font-family: var(--font-family);
  font-weight: lighter;
  font-size: 14px;
  margin-bottom: 10px;

  a:link {
    color: var(--amazon-orange);
    text-decoration: none;
  }

  a:hover {
    color: var(--light-amazon-orange);
    text-decoration: underline;
  }
  
  a:active {
    color: var(--button-click);
    text-decoration: underline;
  }

  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: baseline;
`;
