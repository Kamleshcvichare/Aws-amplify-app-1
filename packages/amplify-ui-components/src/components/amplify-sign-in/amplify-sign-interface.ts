export interface AmplifyForgotPasswordLinkProps {
  forgotPasswordText: string;
  resetPasswordText: string;
}

export interface SignInFormFooterProps {
  submitButtonText: string;
  noAccountText: string;
  createAccountText: string;
  overrideStyle?: boolean;
}