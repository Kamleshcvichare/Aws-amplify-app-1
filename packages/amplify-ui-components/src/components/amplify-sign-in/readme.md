# amplify-sign-in



<!-- Auto Generated Below -->


## Properties

| Property           | Attribute           | Description | Type                   | Default     |
| ------------------ | ------------------- | ----------- | ---------------------- | ----------- |
| `handleSubmit`     | --                  |             | `(Event: any) => void` | `undefined` |
| `validationErrors` | `validation-errors` |             | `string`               | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
