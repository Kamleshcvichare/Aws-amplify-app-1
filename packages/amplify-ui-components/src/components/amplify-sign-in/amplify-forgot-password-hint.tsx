import { FunctionalComponent as FC, h } from '@stencil/core';
import { AmplifyForgotPasswordHintProps } from './amplify-sign-in-interface';
import { AuthStateTunnel } from '../../data/auth-state';

export const AmplifyForgotPasswordHint: FC<AmplifyForgotPasswordHintProps> = ({
  forgotPasswordText,
  resetPasswordText,
}) => (
  <AuthStateTunnel.Consumer>
    {({ onAuthStateChange }) => (
      <div>
        {forgotPasswordText}{' '}
        <amplify-link onClick={() => onAuthStateChange('forgotpassword')}>{resetPasswordText}</amplify-link>
      </div>
    )}
  </AuthStateTunnel.Consumer>
);
