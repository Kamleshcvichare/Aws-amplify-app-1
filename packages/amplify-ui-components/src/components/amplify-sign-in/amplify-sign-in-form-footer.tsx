import { FunctionalComponent as FC, h } from '@stencil/core';
import { AmplifySignInFormFooterProps } from './amplify-sign-in-interface';
import { signInFormFooter } from './amplify-sign-in.styles';

export const AmplifySignInFormFooter: FC<AmplifySignInFormFooterProps> = ({ submitButtonText, noAccountText, createAccountText, overrideStyle = false, onAuthStateChange, authState }) => (
  <div class={signInFormFooter}>
    <span>{noAccountText} <amplify-link onClick={() => onAuthStateChange('signup', authState)}>{createAccountText}</amplify-link></span>
    <amplify-button type="submit" overrideStyle={overrideStyle}>{submitButtonText}</amplify-button>
  </div>
);
