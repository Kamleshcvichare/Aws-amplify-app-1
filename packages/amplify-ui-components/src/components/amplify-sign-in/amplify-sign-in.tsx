import { Component, Prop, State, h } from '@stencil/core';
import { FormFieldTypes } from '../../components/amplify-auth-fields/amplify-auth-fields-interface';
import { AmplifyForgotPasswordHint } from './amplify-forgot-password-hint';
import { AmplifySignInFormFooter } from './amplify-sign-in-form-footer';
import { AuthState } from '../../common/types/auth-types';

import {
  HEADER_TEXT,
  SUBMIT_BUTTON_TEXT,
  CREATE_ACCOUNT_TEXT,
  NO_ACCOUNT_TEXT,
  FORGOT_PASSWORD_TEXT,
  RESET_PASSWORD_TEXT,
} from '../../common/constants';

import { Logger } from '@aws-amplify/core';
import { Auth } from '@aws-amplify/auth';

const logger = new Logger('SignIn');

@Component({
  tag: 'amplify-sign-in',
  shadow: false,
})
export class AmplifySignIn {
  /** Fires when sign in form is submitted */
  @Prop() handleSubmit: (Event) => void = event => this.signIn(event);
  /** Engages when invalid actions occur, such as missing field, etc. */
  @Prop() validationErrors: string;
  /** Used for header text in sign in component */
  @Prop() headerText: string = HEADER_TEXT;
  /** Used for the submit button text in sign in component */
  @Prop() submitButtonText: string = SUBMIT_BUTTON_TEXT;
  /** (Optional) Overrides default styling */
  @Prop() overrideStyle: boolean = false;

  @Prop() handleAuthStateChange: (nextAuthState: AuthState, data?: object) => void;
  /**
   * Form fields allows you to utilize our pre-built components such as username field, code field, password field, email field, etc.
   * by passing an array of strings that you would like the order of the form to be in. If you need more customization, such as changing
   * text for a label or adjust a placeholder, you can follow the structure below in order to do just that.
   * ```
   * [
   *  {
   *    type: 'username'|'password'|'email'|'code'|'default',
   *    label: string,
   *    placeholder: string,
   *    hint: string | Functional Component | null,
   *    required: boolean
   *  }
   * ]
   * ```
   */
  @Prop() formFields: FormFieldTypes | string[];

  /* Whether or not the sign-in component is loading */
  @State() loading: boolean = false;
  /* The username value in the sign-in form */
  @State() username: string;
  /* The password value in the sign-in form */
  @State() password: string;

  componentWillLoad() {
    this.formFields = [
      {
        type: 'username',
        required: true,
        handleInputChange: event => this.handleUsernameChange(event),
      },
      {
        type: 'password',
        hint: (
          <AmplifyForgotPasswordHint
            forgotPasswordText={FORGOT_PASSWORD_TEXT}
            resetPasswordText={RESET_PASSWORD_TEXT}
          />
        ),
        required: true,
        handleInputChange: event => this.handlePasswordChange(event),
      },
    ];
  }

  handleUsernameChange(event) {
    this.username = event.target.value;
    console.log(this.username);
  }

  handlePasswordChange(event) {
    this.password = event.target.value;
  }

  checkContact(user) {
    if (!Auth || typeof Auth.verifiedContact !== 'function') {
      throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
    }
    Auth.verifiedContact(user).then(data => {
      if (Object.keys(data.verified).length === 0) {
        this.handleAuthStateChange(AuthState.SignedIn, user);
      } else {
        user = Object.assign(user, data);
        this.handleAuthStateChange(AuthState.VerifyContact, user);
      }
    });
  }

  async signIn(event) {
    console.log(event);
    // avoid submitting the form
    if (event) {
      event.preventDefault();
    }

    if (!Auth || typeof Auth.signIn !== 'function') {
      throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
    }
    this.loading = true;
    try {
      const user = await Auth.signIn(this.username, this.password);
      logger.debug(user);
      if (user.challengeName === 'SMS_MFA' || user.challengeName === 'SOFTWARE_TOKEN_MFA') {
        logger.debug('confirm user with ' + user.challengeName);
        this.handleAuthStateChange(AuthState.ConfirmSignIn, user);
      } else if (user.challengeName === 'NEW_PASSWORD_REQUIRED') {
        logger.debug('require new password', user.challengeParam);
        this.handleAuthStateChange(AuthState.ResetPassword, user);
      } else if (user.challengeName === 'MFA_SETUP') {
        logger.debug('TOTP setup', user.challengeParam);
        this.handleAuthStateChange(AuthState.TOTPSetup, user);
      } else if (
        user.challengeName === 'CUSTOM_CHALLENGE' &&
        user.challengeParam &&
        user.challengeParam.trigger === 'true'
      ) {
        logger.debug('custom challenge', user.challengeParam);
        this.handleAuthStateChange(AuthState.CustomConfirmSignIn, user);
      } else {
        this.checkContact(user);
      }
    } catch (error) {
      if (error.code === 'UserNotConfirmedException') {
        logger.debug('the user is not confirmed');
        this.handleAuthStateChange(AuthState.ConfirmSignUp, { username: this.username });
      } else if (error.code === 'PasswordResetRequiredException') {
        logger.debug('the user requires a new password');
        this.handleAuthStateChange(AuthState.ForgotPassword, { username: this.username });
      } else {
        throw new Error(error);
      }
    } finally {
      this.loading = false;
    }
  }

  render() {
    return (
      <amplify-form-section
        headerText={this.headerText}
        overrideStyle={this.overrideStyle}
        handleSubmit={this.handleSubmit}
      >
        <amplify-auth-fields formFields={this.formFields} />
        <div slot="amplify-form-section-footer">
          <AmplifySignInFormFooter
            submitButtonText={this.submitButtonText}
            createAccountText={CREATE_ACCOUNT_TEXT}
            noAccountText={NO_ACCOUNT_TEXT}
            onAuthStateChange={this.handleAuthStateChange}
          />
        </div>
      </amplify-form-section>
    );
  }
}
