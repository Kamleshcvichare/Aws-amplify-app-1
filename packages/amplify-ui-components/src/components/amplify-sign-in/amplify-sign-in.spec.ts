import { newSpecPage } from '@stencil/core/testing';
import { AmplifySignIn } from './amplify-sign-in';

describe('amplify-sign-in spec:', () => {
  describe('Component logic ->', () => {
    let signIn;

    beforeEach(() => {
      signIn = new AmplifySignIn();
    });

    it('should render `handleSubmit` as undefined by default', () => {
      expect(signIn.handleSubmit).toBeUndefined();
    });

    it('should render `validationErrors` as undefined by default', () => {
      expect(signIn.validationErrors).toBeUndefined();
    });

    it('should render `headerText` to `Sign into your account` by default', () => {
      expect(signIn.headerText).toEqual('Sign into your account');
    });

    it('should render `submitButtonText` to `Sign in` by default', () => {
      expect(signIn.submitButtonText).toEqual('Sign in');
    });

    it('should render `overrideStyle` to false by default', () => {
      expect(signIn.overrideStyle).toBe(false);
    });
  });
  describe('Render logic ->', () => {
    it('should render a `sign in` form by default', async () => {
      const page = await newSpecPage({
        components: [AmplifySignIn],
        html: `<amplify-sign-in></amplify-sign-in>`
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});