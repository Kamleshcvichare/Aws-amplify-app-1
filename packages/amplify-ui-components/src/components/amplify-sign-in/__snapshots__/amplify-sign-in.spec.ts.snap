// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-sign-in spec: Render logic -> should render a \`sign in\` form by default 1`] = `
<amplify-sign-in>
  <amplify-form-section headertext="Sign into your account">
    <amplify-auth-fields></amplify-auth-fields>
  </amplify-form-section>
</amplify-sign-in>
`;
